Update dependencies

grzesiek2010 opened the pull request:
#### What has been done to verify that this works as intended?
Reviewed implemented changes, tested the app manually (OSM maps)

#### Why is this the best possible solution? Were any other approaches considered?
It's something we do after each release to keep the dependencies updated.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
Usually we don't need those updates but this time I think it would be good to test OSM maps.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/opendatakit/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [ ] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [ ] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [ ] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/opendatakit/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

grzesiek2010 committed:
Updated dependencies

grzesiek2010 committed:
Upgrading to LeakCanary 2

grzesiek2010 committed:
Updated osmdroid

grzesiek2010 committed:
More updates

grzesiek2010 commented on collect_app/build.gradle:
```diff
@@ -262,15 +262,10 @@ dependencies {
-    releaseImplementation "com.squareup.leakcanary:leakcanary-android-no-op:${rootProject.leakyCanaryVersion}"
-    testImplementation "com.squareup.leakcanary:leakcanary-android-no-op:${rootProject.leakyCanaryVersion}"
-    androidTestImplementation "com.squareup.leakcanary:leakcanary-android-no-op:${rootProject.leakyCanaryVersion}"
-    odkCollectReleaseImplementation "com.squareup.leakcanary:leakcanary-android-no-op:${rootProject.leakyCanaryVersion}"
```
According to https://square.github.io/leakcanary/upgrading-to-leakcanary-2.0/ we no longer need no-op artifacts

seadowg commented on collect_app/src/main/java/org/odk/collect/android/application/TestCollect.java:
```diff
@@ -12,12 +10,6 @@
 public class TestCollect extends Collect {
 
-    @Override
-    protected RefWatcher setupLeakCanary() {
```
Huh so LeakCanary won't be running in any test build automatically now?

grzesiek2010 replied:
That's what I understand https://square.github.io/leakcanary/recipes/#running-leakcanary-in-instrumentation-tests

seadowg replied:
So much better!

grzesiek2010 commented:
@seadowg please take a quick look if you want and add `needs testing` label

seadowg committed:
Use Android Library for nbi-stubs rather than precompiled .aar [fixes #3732]

This means the aar used in the build should work with build tools
updates. Should also make it easier to debug any problems as the build
chain will be clearer.

seadowg commented:
Just removing "needs testing" and my review as I'm going to a push a fix for the gradle plugin problem.

seadowg committed:
Correct typo comment

grzesiek2010 commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -1,9 +1,8 @@
 android.enableD8.desugaring=true
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.3-all.zip
-# Android Studio Gradle wrapper >= 3.1.2 doesn't support configure on demand
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.4-all.zip
```
Why not the latest 6.2.2?

seadowg replied:
Huh that was auto generated. Probably just game me the minimum for that plugin version. I'll update it.

seadowg replied:
Wow this is causing all kinds of problems. Did you try and update spotbugs at all as part of this? It looks like we might be using a very old version.

grzesiek2010 replied:
ahh spotbugs... I think I was able to update it to version 2.0.0 but not the latest 4.0.4 because it was related somehow with gradle version

seadowg committed:
Remove spotbugs

seadowg committed:
Move bikram-sambat dep to local jar to prevent Gradle 6.2.2 problems

seadowg committed:
Update Gradle

seadowg commented on build.gradle:
```diff
@@ -23,7 +23,6 @@ allprojects {
         jcenter()
         maven { url 'https://oss.sonatype.org/content/groups/public' }
         maven { url 'https://jitpack.io' }
-        maven { url 'https://staging.dev.medicmobile.org/_couch/maven-repo' }
```
This repo link doesn't seem to be working any longer as they don't provide a `.pom` for the dependency. Given its a pretty stagnant dependency I think using a local jar is fine.

grzesiek2010 replied:
I think such a trick would work:
```
maven {
            url 'https://staging.dev.medicmobile.org/_couch/maven-repo'
            metadataSources { artifact() }
        }
```
and you won't need anything more.

seadowg replied:
I was looking around for something like this as it really surprised me that there wasn't a way to get it working with just a JAR. Will give this a go tomorrow!

lognaturel replied:
Great if we can track the Medic repo but not super important I don’t think.

grzesiek2010 replied:
Yes I was right. Just added a commit on my own, so that you don't need to play with it just take a look.

lognaturel commented:
Lots of good stuff unlocked by the minSDK change. 👍 Agreed that checking OSM would be good. Since there's now also a change to the native architecture stubs, it would be good to also quickly verify Mapbox. I'm going to see if I can also get https://github.com/opendatakit/collect/pull/3457 complete shortly so we can be fully up to date. That will also require Mapbox checking so really here it can just be to sanity check that it works on more than one device.

grzesiek2010 committed:
Added the Medic repo back

grzesiek2010 commented on collect_app/build.gradle:
```diff
@@ -194,12 +196,6 @@ configurations.all {
-
     implementation 'androidx.legacy:legacy-support-v13:1.0.0'
     implementation 'androidx.browser:browser:1.2.0'
     implementation 'androidx.appcompat:appcompat:1.1.0-rc01' // Shouldn't be upgraded until https://issuetracker.google.com/issues/141869006 is fixed
```
It's marked as `won't fix`. We can leave it as if for now but in this case we can't block it for ever.

seadowg replied:
We actully have our own issue for tracking this with notes on the possible workaround. I'll update this so it references that rather than Google's issue.

seadowg committed:
Update comment on appcompat issue

mmarciniak90 commented:
Verified with success on Android: 6.0, 7.0, 8.1, 9.0

I didn't notice regression.
New design of zoom in / out buttons visible.
<img height="40%" width="40%" src="https://user-images.githubusercontent.com/3907554/77891193-e3161400-7270-11ea-9fbc-16f198f1fe75.png">

**Verified cases:**
* Carto, Stamen, USGS, OSM
* Mapbox
* OSM on view showing submissions on map
* Message at the bottom of the map
* Geopoint, GeoTrace, Geoshape
* 3 input methods 
* zoom in / out
* offline layer
* toast with not enough number of point to create Polyline/Polygon
* English/Spanish language

@kkrawczyk123 take a look at this PR, especially on Android 5.1

kkrawczyk123 commented:
I haven't noticed any regression too. 
Verified on Androids 5.1 and 10.

@opendatakit-bot unlabel "needs testing"
@opendatakit-bot label "behavior verified"
